packages/bridge/src/skip/transfer-status.ts

```diff
@@ -40,39 +60,38 @@ export class SkipTransferStatusProvider implements TransferStatusProvider {
 
     await poll({
       fn: async () => {
-        try {
-          const txStatus = await this.skipClient.transactionStatus({
-            chainID: fromChainId.toString(),
-            txHash: sendTxHash,
+        const tx = {
+          chainID: fromChainId.toString(),
+          txHash: sendTxHash,
+          env: this.env,
+        };
+
+        const txStatus = await this.skipStatusProvider
+          .transactionStatus(tx)
+          .catch(async (error) => {
+            if (error instanceof Error && error.message.includes("not found")) {
+              // if we get an error that it's not found, prompt skip to track it first
+              // then try again
+              await this.skipStatusProvider.trackTransaction(tx);
+              return this.skipStatusProvider.transactionStatus(tx);
```

Comment:
Wouldn't this call short-circuit the `poll` since this is returning a promise without a catch statement?

Reply:
WDYM without a catch statement? I believe what would happen here is it would be transformed into the same promise that is awaited at the initial call above

Reply:
It wouldn’t, since this isn't a recursive call and it's using the dependency-injected method. If we want to share the catch statement, we’d likely need to create an inline function and call it inside the catch method

Reply:
I'm not sure I follow, how would this look?

Reply:
Actually, I think we could avoid recursion by just adding a catch 

```suggestion
              return this.skipStatusProvider.transactionStatus(tx).catch(() => {});
```

Reply:
I don't think recursion is being used, we're just calling the same given function twice